### PR TITLE
CMR-8803 Adding in Update/Append/Remove for BrowseImageURL as well as…

### PR DESCRIPTION
**Pull request by eereiter**
… for online resource and online access urls, since they are sharing the same code.

### CODE REVIEW

File: ingest-app/docs/api.md [hunk 1]
```diff
@@ -1822,16 +1803,149 @@ curl -XPOST \
...
-        {
-            "GranuleUR": "my-gran-ur",
-            "Links": [
-                { "from": "https://old-link", "to": "http://new-link" }
-            ]
-        }
-    ]
+    "updates": [{"GranuleUR": "my-gran-ur",
+                 "Links": [{ "from": "https://old-link", "to": "http://new-link" }]}]
+}'
```
Comment (sxu123) [inline ingest-app/docs/api.md:1808]: From the examples given, seems that just use {"URL": "http://update-to-new-link.com"} is cleaner, than "from" "to".

Comment (eereiter) [reply ingest-app/docs/api.md:1808]: I need to have the from and to.  The from lets the code know which of the related urls, online resources, online access urls, or image urls I need to replace out of the many that can exist.  The to: the code needs to know what to replace it with.

**Comment by sxu123** [comment]
Is It possible that you could add some integration tests? I only saw unit tests and I don't know if it all works together because some part of the code is not so easy to understand.

**Comment by eereiter** [comment]
I have added tests, for the code that I have changed.  The code that I changed only involves updating echo 10 records.  
There are a lot of existing system integration tests that I don't think we need to duplicate:  These tests can be found here: 
system-int-test/test/cmr/system_int_test/ingest/granule_bulk_update/granule_bulk_update_endpoint_test.clj
system-int-test/test/cmr/system_int_test/ingest/granule_bulk_update/granule_bulk_update_flow_test.clj
system-int-test/test/cmr/system_int_test/ingest/granule_bulk_update/granule_bulk_update_schema_test.clj
system-int-test/test/cmr/system_int_test/ingest/granule_bulk_update/granule_bulk_update_test.clj
system-int-test/test/cmr/system_int_test/ingest/granule_bulk_update/granule_bulk_update_volume_test.clj
system-int-test/test/cmr/system_int_test/ingest/granule_bulk_update/mime_type_test.clj
system-int-test/test/cmr/system_int_test/ingest/granule_bulk_update/online_resource_url_test.clj

Then there are many unit tests that test a lot of the updates and adds to the actual ECHO 10 or UMM-G records:
ingest-app/test/cmr/ingest/services/granule_bulk_update/additional_file/umm_g_test.clj
ingest-app/test/cmr/ingest/services/granule_bulk_update/checksum_size_format/echo10_test.clj
ingest-app/test/cmr/ingest/services/granule_bulk_update/checksum_size_format/size_input_test.clj
ingest-app/test/cmr/ingest/services/granule_bulk_update/mime_type/echo10_test.clj
ingest-app/test/cmr/ingest/services/granule_bulk_update/mime_type/umm_g_test.clj
ingest-app/test/cmr/ingest/services/granule_bulk_update/opendap/echo10_test.clj
ingest-app/test/cmr/ingest/services/granule_bulk_update/opendap/opendap_util_test.clj
ingest-app/test/cmr/ingest/services/granule_bulk_update/opendap/umm_g_test.clj
ingest-app/test/cmr/ingest/services/granule_bulk_update/s3/echo10_test.clj
ingest-app/test/cmr/ingest/services/granule_bulk_update/s3/s3_util_test.clj
ingest-app/test/cmr/ingest/services/granule_bulk_update/s3/umm_g_test.clj
ingest-app/test/cmr/ingest/services/granule_bulk_update_service_test.clj
ingest-app/test/cmr/ingest/test/data/granule_bulk_update.clj

**Review by sxu123 — APPROVED** [review]

**Review by eudoroolivares2016 — APPROVED** [review]

**Review by zimzoom — APPROVED** [review]

**Pull request merged**